lib/class-wp-theme-json-gutenberg.php

```diff
@@ -2425,8 +2461,34 @@ function( $pseudo_selector ) use ( $selector ) {
 		$declarations_duotone = array();
 		foreach ( $declarations as $index => $declaration ) {
 			if ( 'filter' === $declaration['name'] ) {
-				unset( $declarations[ $index ] );
+				$filter_preset = $this->gutenberg_get_preset_from_declaration( $declarations[ $index ] );
+				$filter_svg = gutenberg_get_duotone_filter_svg( $filter_preset );
+
+				add_action(
+					'wp_footer',
+					static function () use ( $filter_svg, $selector ) {
+						echo $filter_svg;
+
+						/*
+						 * Safari renders elements incorrectly on first paint when the
+						 * SVG filter comes after the content that it is filtering, so
+						 * we force a repaint with a WebKit hack which solves the issue.
+						 */
+						global $is_safari;
+						if ( $is_safari ) {
+							/*
+							 * Simply accessing el.offsetHeight flushes layout and style
+							 * changes in WebKit without having to wait for setTimeout.
+							 */
+							printf(
+								'<script>( function() { var el = document.querySelector( %s ); var display = el.style.display; el.style.display = "none"; el.offsetHeight; el.style.display = display; } )();</script>',
+								wp_json_encode( $selector )
+							);
```

Comment:
It would be nice if we could avoid the Safari hack in as many cases as possible. It exists to fix this problem. (#37954)

![broken safari duotone](https://lendedotblog.files.wordpress.com/2023/01/duotone-safari.png)